### PR TITLE
[feat] evm shop

### PR DESCRIPTION
**Pull request by vanminh1701**

### CODE REVIEW

File: src/constant/wallet.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  Solflare = 'Solflare',
+  Metamask = 'Metamask',
+}
+
+enum Network {
+  SolDevnet = 'Devnet',
+  SolMainnetBeta = 'Mainnet Beta',
+  Eth = 'Eth',
+  EthTestnet = 'Goerli',
+  Polygon = 'Polygon matic',
```
Comment (RustySol) [inline src/constant/wallet.ts]: Polygon Matic

File: src/constant/wallet.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  PolygonTestnet = 'Polygon mumbai',
```
Comment (RustySol) [inline src/constant/wallet.ts]: Polygon Mumbai

File: src/views/MarketplaceWithFilter.tsx [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
-        description={
-          "Enable users to filter by collections and search for NFTs."
-        }
+        title={'Marketplace'}
+        description={'Enable users to filter by collections and search for NFTs.'}
         style={{ paddingBottom: 50 }}
       />
       <Orders
-        wallet={wallet}
+        wallet={wallet as AnchorWallet}
```
Comment (RustySol) [inline src/views/MarketplaceWithFilter.tsx]: how about EVM wallets?

File: src/hooks/useUserWallet.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+export const useUserWallet = () => {
+  const [wallet, network] = useWalletStore((s) => [s.wallet, s.network]);
+
+  return useMemo<AnchorWallet | EthWallet | undefined>(() => {
+    if (!wallet?.publicKey) return undefined;
+    if (getBlockchain(network) === BlockchainType.SOL) return wallet as AnchorWallet;
+    return {
+      publicKey: wallet?.publicKey.toString(),
+      web3Modal: getWeb3Modal(),
```
Comment (RustySol) [inline src/hooks/useUserWallet.ts:16]: is `web3Modal` still needed here?

Comment (vanminh1701) [reply src/hooks/useUserWallet.ts:16]: yes, for evm, we're still using web3Modal to connect metamask wallet

File: src/utils/evm-networks.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
+import { message } from 'antd';
+import { Blockchain } from '@liqnft/candy-shop-types';
+
+export enum ChainId {
+  MATIC = 137,
+  MUMBAI = 80001,
+  ETH = 1,
+  GOERLI = 420,
```
Comment (RustySol) [inline src/utils/evm-networks.ts]: `5`
ref: https://besu.hyperledger.org/en/stable/public-networks/concepts/network-and-chain-id/

File: src/utils/evm-networks.ts [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+      decimals: 18,
+      symbol: 'ETH',
+    },
+    rpcUrls: ['https://polygontestapi.terminet.io/rpc'],
+  },
+  [ChainId.GOERLI]: {
+    chainName: 'Optimism Goerli Testnet',
+    chainId: '0x' + ChainId.GOERLI.toString(16),
+    nativeCurrency: {
+      name: 'Optimism Goerli Testnet',
```
Comment (RustySol) [inline src/utils/evm-networks.ts]: just `Goerli` Testnet

**Comment by vanminh1701** [comment]
To make the project works with both EVM and SOL blockchain, I have to rewrite the way we get user wallet when connecting to supported wallet: SOL (Phantom, SolFlare), EVM (Metamask).
1. new UI `ConnectButton`: handle connection on multiple chains, it is using hook `useConnection` to generate custom user wallet interface
2. new UI `WalletModal`: the UI modal for selecting which wallet user will connect 
3. new hook `useUserWallet`: to get wallet and pass it to CandyShop UI component
4. global store `useShopStore`: using `zustand` to create a global store for shop config
5. evm-nework.ts: supported network configuration for EVM 
6. add prettier config for better sort import config
7. rename some env variable names
```
REACT_APP_SOLANA_NETWORK => REACT_APP_NETWORK
REACT_APP_SOLANA_RPC_HOST => REACT_APP_RPC_HOST
```

File: src/views/MarketplaceWithFilter.tsx (hunk 3 above)
Comment (RustySol) [reply src/views/MarketplaceWithFilter.tsx]: > the current UI is support only Marketplace and Sell, so other view will use as AnchorWallet
image

**Comment by haVincy** [comment]
> To make the project works with both EVM and SOL blockchain, I have to rewrite the way we get user wallet when connecting to supported wallet: SOL (Phantom, SolFlare), EVM (Metamask).
> 
> 1. new UI `ConnectButton`: handle connection on multiple chains, it is using hook `useConnection` to generate custom user wallet interface
> 2. new UI `WalletModal`: the UI modal for selecting which wallet user will connect
> 3. new hook `useUserWallet`: to get wallet and pass it to CandyShop UI component
> 4. global store `useShopStore`: using `zustand` to create a global store for shop config
> 5. evm-nework.ts: supported network configuration for EVM
> 6. add prettier config for better sort import config
> 7. rename some env variable names
> 
> ```
> REACT_APP_SOLANA_NETWORK => REACT_APP_NETWORK
> REACT_APP_SOLANA_RPC_HOST => REACT_APP_RPC_HOST
> ```

This is great, @vanminh1701, it would be helpful if we can add those new UI components with code snippet to introduce it to outside devs in Wiki like your comment here
- https://github.com/LIQNFT/candy-shop-storefront/wiki

**Review by haVincy — APPROVED** [review]

**Pull request merged**